### ISSUE TITLE
docker: Error response from daemon: failed to verify image

### ISSUE THREAD

**Issue by tsuna** [moby/moby#23477]
**Output of `docker version`:**

```
core@r12s6 ~ $ docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   8acee1b
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   8acee1b
 Built:        
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.10.3
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.5.0-coreos-r1
Operating System: CoreOS 1010.5.0 (MoreOS)
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 62.95 GiB
Name: r12s6.sjc.aristanetworks.com
ID: HCTY:Q3PK:PF2E:5XDO:RNWV:GYGS:LD5E:GVC5:Q3KC:7ED5:I2HL:GAMZ
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Fresh install of CoreOS stable (1010.5.0) on a dedicated server.

**Steps to reproduce the issue:**
I don't know why but on this particular host, after reformatting the root filesystem and doing a fresh CoreOS install, I'm unable to deploy this particular container:

```
core@r12s6 ~ $ docker run --restart=always --net=host --name=lldpd -d registry.docker.sjc.aristanetworks.com:5000/tools/lldpd
docker: Error response from daemon: failed to verify image: sha256:6720cea9f393cd435c53b7ac3d4b8ac8a60e0341308251d52bcb8a8911263fc0.
See 'docker run --help'.
```

**Describe the results you received:**

Error above.

**Describe the results you expected:**

Container should start.

**Additional information you deem important (e.g. issue happens only occasionally):**
I did a `rm -rf /var/lib/docker/image/overlay/layerdb/sha256/*` but that didn't change anything.

**Comment by justincormack** [comment]
Are there any more error messages in the daemon logs? It looks like you might have a corrupted image file, although it is not entirely clear from the error message.

**Comment by tsuna** [comment]
The relevant log seems to be:

```
core@r12s6 ~ $ journalctl -u docker.service | tail -n 2
Jun 13 09:36:50 r12s6.sjc.aristanetworks.com dockerd[1130]: time="2016-06-13T09:36:50.094817415Z" level=warning msg="HostsPath set to \"/var/lib/docker/containers/b835d568452d477dcdb9d1aad67bae065ba888257db7c01c2d62467690f8b308/hosts\", but can't stat this filename (err = stat /var/lib/docker/containers/b835d568452d477dcdb9d1aad67bae065ba888257db7c01c2d62467690f8b308/hosts: no such file or directory); skipping"
Jun 13 09:36:55 r12s6.sjc.aristanetworks.com dockerd[1130]: time="2016-06-13T09:36:55.293853797Z" level=error msg="Handler for POST /v1.22/containers/create returned error: failed to verify image: sha256:6720cea9f393cd435c53b7ac3d4b8ac8a60e0341308251d52bcb8a8911263fc0"
```

**Comment by tsuna** [comment]
I ran into this issue again on another host after upgrading to CoreOS 1010.5.0 – the only solution I found was to `rm -rf` everything under `/var/lib/docker` and then `systemctl restart docker.service`.  In this second instance, the machine unexpectedly crash-rebooted while the first container got deployed on it.  I can't tell why it crashed (kernel panic or what) but upon rebooting then Docker was unable to use the image of that first container until I cleared everything under `/var/lib/docker`.

What sort of debugging information can I provide to help nail down why the docker daemon gets stuck in this unhappy state for that particular image?
